src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duke.util;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles file reading and writing.
+ */
+public class Storage {
+
+    private File storage;
+    private String filepath;
+
+    /**
+     * Initialises a Storage object for file processes.
+     * @param filepath path to the file for storage
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+        storage = new File(filepath);
+    }
+
+    /**
+     * Loads data from the file.
+     * @return ArrayList of Task objects
+     * @throws DukeException if file is not found
+     */
+    public ArrayList<Task> load() throws DukeException {
```

Comment:
Minor suggestion, would it better implement 'programming to an interface' methodology if the code worked with the base interface rather than the implementing class wherever possible? i.e.
```suggestion
    public List<Task> load() throws DukeException {
```